balena-os/meta-balena
# Add oneshot service to migrate supervisor state config

The hostapp-update hooks would run before the supervisor update step
when using balenahup via the dashboard. This resulted in the balena
supervisor state conf file being out-of-date.

This commit removes the hostapp-update hook and instead uses a
oneshot service to ensure the conf is migrated if it does not exist
in the new path before starting the supervisor.

Resolves: https://github.com/balena-os/meta-balena/issues/2201, https://github.com/balena-os/balenahup/issues/347
Change-type: patch
Signed-off-by: Kyle Harding <kyle@balena.io>


**Manual test case**

1. Provision a Pi4 to 2.75.0+rev1
2. Perform HUP via the staging dashboard to an image from this PR
3. After booting in the new OS check if supervisor is running and check journal logs

**Automated test case**

HUP test cases are WIP here: https://github.com/balena-os/meta-balena/pull/2174

---
### Contributor checklist



### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
